builder/builder.go

```diff
@@ -163,15 +163,15 @@ func (b *Builder) Build(payload *Payload) error {
 	})
 	// Index txs.
 	if err := b.txStore.Add(txResults); err != nil {
-		return fmt.Errorf("add tx results: %v", err)
+		return nil, fmt.Errorf("add tx results: %v", err)
 	}
 	// Publish events.
 	for _, txResult := range txResults {
 		if err := b.eventBus.PublishEventTx(bfttypes.EventDataTx{
 			TxResult: *txResult,
 		}); err != nil {
-			return fmt.Errorf("publish event tx: %v", err)
+			return nil, fmt.Errorf("publish event tx: %v", err)
 		}
 	}
-	return nil
+	return b.blockStore.HeadBlock(), nil
```

Comment:
Let's build the struct here to avoid hitting the db.

Reply:
194b0f8